var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -93,26 +93,19 @@ def url_for_version(self, version):
 
     # High-level API of HDF5 1.8.9 or later is required for netCDF-4 support:
     # http://www.unidata.ucar.edu/software/netcdf/docs/getting_and_building_netcdf.html
-    depends_on('hdf5@1.8.9:+hl~mpi', when='~mpi')
-    depends_on('hdf5@1.8.9:+hl+mpi', when='+mpi')
+    depends_on('hdf5@1.8.9:+hl')
 
     # Starting version 4.4.0, it became possible to disable parallel I/O even
     # if HDF5 supports it. For previous versions of the library we need
-    # HDF5 without mpi support to disable parallel I/O.
-    # The following doesn't work if hdf5+mpi by default and netcdf-c~mpi is
-    # specified in packages.yaml
+    # HDF5 without mpi support to disable parallel I/O:
     # depends_on('hdf5~mpi', when='@:4.3~mpi')
-    # Thus, we have to introduce a conflict
-    conflicts('~mpi', when='@:4.3^hdf5+mpi',
-              msg='netcdf-c@:4.3~mpi requires hdf5~mpi')
+    # However, it is very unlikely that someone needs such combination.
+    # Therefore, we put the constraint for all versions to be able to avoid the
+    # dependency on MPI by simply running: spack install netcdf-c~mpi.
+    depends_on('hdf5~mpi', when='~mpi')
 
     # We need HDF5 with mpi support to enable parallel I/O.
-    # The following doesn't work if hdf5~mpi by default and netcdf-c+mpi is
-    # specified in packages.yaml
-    # depends_on('hdf5+mpi', when='+mpi')
-    # Thus, we have to introduce a conflict
-    conflicts('+mpi', when='^hdf5~mpi',
-              msg='netcdf-c+mpi requires hdf5+mpi')
+    depends_on('hdf5+mpi', when='+mpi')
```

Comment:
This is what is left from the splitting of the constraints for the case `when='+mpi'`. Again, we don't' need the conflict anymore because it is basically impossible to have `netcdf-c+mpi ^hdf5~mpi` due to the `depends_on` statements.